tidb-binlog/pump_client/pump.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package client
+
+import (
+	"net"
+	"time"
+
+	"github.com/ngaut/log"
+	"github.com/pingcap/tidb-tools/tidb-binlog/node"
+	pb "github.com/pingcap/tipb/go-binlog"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+// PumpStatus saves pump's status
+type PumpStatus struct {
+	node.Status
+
+	// the pump is avaliable or not.
+	IsAvaliable bool
+
+	// the client of this pump
+	Client pb.PumpClient
+}
+
+// NewPumpStatus returns a new PumpStatus according to node's status.
+func NewPumpStatus(status *node.Status) *PumpStatus {
+	pumpStatus := &PumpStatus{}
+	pumpStatus.Status = *status
+	pumpStatus.IsAvaliable = (status.State == node.Online)
+
+	if !pumpStatus.IsAvaliable {
```

Comment:
do we need it? maybe we dont need check isAvaliable, as we said,  is `online` enough?